modules/cli/src/main/scala/scala/cli/commands/Repl.scala

```diff
@@ -183,14 +186,15 @@ object Repl extends ScalaCommand[ReplOptions] {
         options.javaHome().value.javaCommand,
         replArtifacts.replJavaOpts ++ options.javaOptions.javaOpts.toSeq.map(_.value.value),
         classDir.map(_.toIO).toSeq ++ replArtifacts.replClassPath.map(_.toIO),
-        replArtifacts.replMainClass,
-        if (Properties.isWin)
-          options.notForBloopOptions.replOptions.ammoniteArgs.map { a =>
-            if (a.contains(" ")) "\"" + a.replace("\"", "\\\"") + "\""
-            else a
-          }
-        else
-          options.notForBloopOptions.replOptions.ammoniteArgs,
+        replArtifacts.replMainClass, {
+          if (Properties.isWin)
+            options.notForBloopOptions.replOptions.ammoniteArgs.map { a =>
+              if (a.contains(" ")) "\"" + a.replace("\"", "\\\"") + "\""
+              else a
+            }
+          else
+            options.notForBloopOptions.replOptions.ammoniteArgs
+        } ++ programArgs,
```

Comment:
We probably need to do the `if (a.contains(" ")) "\"" + a.replace("\"", "\\\"") + "\""` stuff to what's in `programArgs` too. Better factor `options.notForBloopOptions.replOptions.ammoniteArgs ++ programArgs` a bit before, then do the `if (Properties.isWin) …` stuff on the result.

Reply:
It's make sense, good point